packages/tools/testing-tools/test/saga.test.ts

```diff
@@ -14,261 +20,518 @@ afterAll(() => {
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteCommand({
+      type: 'create',
+      aggregateName: 'user',
+      aggregateId: 'id',
+      payload: {
+        item: 'aggregate-id',
+      },
+    })
+})
+
+test('shouldExecuteQuery assertion', async () => {
+  await givenEvents([
+    {
+      type: 'Query',
+      aggregateId: 'aggregate-id',
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteQuery({
+      modelName: 'model',
+      resolverName: 'test',
+      resolverArgs: { test: 'test' },
+      jwt: 'user',
+    })
+})
 
-  test('shouldExecuteSideEffect & shouldExecuteCommand', async () => {
+test('shouldExecuteCommand & shouldExecuteQuery simultaneously', async () => {
+  await givenEvents([
+    {
+      type: 'CommandQuery',
+      aggregateId: 'aggregate-id',
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteCommand({
+      type: 'create',
+      aggregateName: 'user',
+      aggregateId: 'id',
+      payload: {
+        item: 'aggregate-id',
+      },
+    })
+    .shouldExecuteQuery({
+      modelName: 'model',
+      resolverName: 'test',
+      resolverArgs: { test: 'test' },
+      jwt: 'user',
+    })
+})
+
+test('shouldExecuteSideEffect', async () => {
+  await givenEvents([
+    {
+      type: 'SideEffect',
+      aggregateId: 'aggregate-id',
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteSideEffect('email', 'test', 'aggregate-id')
+})
+
+test('shouldExecuteSideEffect & shouldExecuteCommand', async () => {
+  await givenEvents([
+    {
+      type: 'CommandSideEffect',
+      aggregateId: 'aggregate-id',
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteSideEffect('email', 'test', 'aggregate-id')
+    .shouldExecuteCommand({
+      type: 'create',
+      aggregateName: 'user',
+      aggregateId: 'id',
+      payload: {
+        item: 'aggregate-id',
+      },
+    })
+})
+
+test('shouldExecuteSideEffect & shouldExecuteQuery', async () => {
+  await givenEvents([
+    {
+      type: 'QuerySideEffect',
+      aggregateId: 'aggregate-id',
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteSideEffect('email', 'test', 'aggregate-id')
+    .shouldExecuteQuery({
+      modelName: 'model',
+      resolverName: 'test',
+      resolverArgs: { test: 'test' },
+      jwt: 'user',
+    })
+})
+
+test('shouldExecuteSideEffect, shouldExecuteCommand & shouldExecuteQuery', async () => {
+  await givenEvents([
+    {
+      type: 'CommandQuerySideEffect',
+      aggregateId: 'aggregate-id',
+    },
+  ])
+    .saga(saga)
+    .shouldExecuteSideEffect('email', 'test', 'aggregate-id')
+    .shouldExecuteCommand({
+      type: 'create',
+      aggregateName: 'user',
+      aggregateId: 'id',
+      payload: {
+        item: 'aggregate-id',
+      },
+    })
+    .shouldExecuteQuery({
+      modelName: 'model',
+      resolverName: 'test',
+      resolverArgs: { test: 'test' },
+      jwt: 'user',
+    })
+})
+
+test('shouldExecuteCommand failure output should contain all executed commands', async () => {
+  try {
```

Comment:
In that way we will never know if shouldExecuteCommand method stops to fail. And the same for the next two tests.